rca/enquire_to_study/views.py

```diff
@@ -18,22 +52,112 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
     def post_mailchimp(self):
-        # see https://git.torchbox.com/nesta/nesta-wagtail/-/blob/master/nesta/mailchimp/api.py
         pass
 
+    def get_qs_data(self, query):
+        return requests.get(
+            f"{settings.QS_API_ENDPOINT}/{query}",
+            auth=(settings.QS_API_USERNAME, settings.QS_API_PASSWORD),
+        ).json()
+
     def post_qs(self, form_data):
-        pass
+        # Get data from QS student enquiry endpoint for matching up selected
+        # programmes and programme types
+        qs_level_studies = self.get_qs_data(query="levelofstudies")
+        qs_courses = self.get_qs_data(query="courses")
+
+        data = {
+            "FirstName": form_data["first_name"],
+            "LastName": form_data["last_name"],
+            "EmailAddress": form_data["email"],
+            "MobileNumber": f"{form_data['phone_number'].country_code}||{form_data['phone_number'].national_number}",
+            "subscribedToDirectEmails": form_data["is_notification_opt_in"],
+            "subscribedToDirectPhoneCalls": False,
+            "subscribedToDirectSMS": False,
+            "Source": "RCA Website Webform",
+            "Intake": form_data["start_date"].qs_code,
+            "Notes": f"Enquiry reason: {form_data['enquiry_reason'].reason}",
+            "tags": [
+                {
+                    "TagType": "Source",
+                    "Tag": "rca_wf_xx_RCA Website Webform_xx_xx_xx-xx-xx",
+                }
+            ],
+            "consents": [
+                {
+                    "Consent": form_data["is_read_data_protection_policy"],
+                    "ConsentType": "is_read_data_protection_policy",
+                }
+            ],
+        }
+
+        # USE IOC format to send data, eg CA (Canada) Should be CAN
+        for k, v in IOC_TO_ISO.items():
+            if v == form_data["country_of_citizenship"]:
+                data["CountryOfCitizenship"] = k
+            if v == form_data["country_of_residence"]:
+                data["CountryOfResidence"] = k
+
+        selected_level_of_study_list = []
+        selected_course_list = []
+        for programme in form_data["programmes"]:
+            course_code = next(
+                (
+                    course
+                    for course in qs_courses
+                    if course["codeExternal"] == str(programme.qs_code)
+                ),
+                None,
+            )
+            if not course_code:
+                raise Exception(f"{programme.title} not found in QS course list")
```

Comment:
Would it be worth raising a more specific exception, maybe `ValueError`? https://docs.python.org/3/library/exceptions.html